tests/components/accuweather/test_weather.py

```diff
@@ -0,0 +1,155 @@
+    ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_BEARING,
+    ATTR_FORECAST_WIND_SPEED,
+    ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_OZONE,
+    ATTR_WEATHER_PRESSURE,
+    ATTR_WEATHER_TEMPERATURE,
+    ATTR_WEATHER_VISIBILITY,
+    ATTR_WEATHER_WIND_BEARING,
+    ATTR_WEATHER_WIND_SPEED,
+)
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_ENTITY_ID, STATE_UNAVAILABLE
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from tests.async_mock import patch
+from tests.common import async_fire_time_changed, load_fixture
+from tests.components.accuweather import init_integration
+
+
+async def test_weather_without_forecast(hass):
+    """Test states of the weather without forecast."""
+    await init_integration(hass)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("weather.home")
+    assert state
+    assert state.state == "sunny"
+    assert not state.attributes.get(ATTR_FORECAST)
+    assert state.attributes.get(ATTR_WEATHER_HUMIDITY) == 67
+    assert not state.attributes.get(ATTR_WEATHER_OZONE)
+    assert state.attributes.get(ATTR_WEATHER_PRESSURE) == 1012.0
+    assert state.attributes.get(ATTR_WEATHER_TEMPERATURE) == 22.6
+    assert state.attributes.get(ATTR_WEATHER_VISIBILITY) == 16.1
+    assert state.attributes.get(ATTR_WEATHER_WIND_BEARING) == 180
+    assert state.attributes.get(ATTR_WEATHER_WIND_SPEED) == 14.5
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+
+    entry = registry.async_get("weather.home")
+    assert entry
+    assert entry.unique_id == "0123456"
+
+
+async def test_weather_with_forecast(hass):
+    """Test states of the weather with forecast."""
+    await init_integration(hass, forecast=True)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("weather.home")
+    assert state
+    assert state.state == "sunny"
+    assert state.attributes.get(ATTR_WEATHER_HUMIDITY) == 67
+    assert state.attributes.get(ATTR_WEATHER_OZONE) == 32
+    assert state.attributes.get(ATTR_WEATHER_PRESSURE) == 1012.0
+    assert state.attributes.get(ATTR_WEATHER_TEMPERATURE) == 22.6
+    assert state.attributes.get(ATTR_WEATHER_VISIBILITY) == 16.1
+    assert state.attributes.get(ATTR_WEATHER_WIND_BEARING) == 180
+    assert state.attributes.get(ATTR_WEATHER_WIND_SPEED) == 14.5
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    forecast = state.attributes.get(ATTR_FORECAST)[0]
+    assert forecast.get(ATTR_FORECAST_CONDITION) == "lightning-rainy"
+    assert forecast.get(ATTR_FORECAST_PRECIPITATION) == 4.8
+    assert forecast.get(ATTR_FORECAST_PRECIPITATION_PROBABILITY) == 58
+    assert forecast.get(ATTR_FORECAST_TEMP) == 29.5
+    assert forecast.get(ATTR_FORECAST_TEMP_LOW) == 15.4
+    assert forecast.get(ATTR_FORECAST_TIME) == "2020-07-26T05:00:00+00:00"
+    assert forecast.get(ATTR_FORECAST_WIND_BEARING) == 166
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)